client/handlers.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/mikloslorinczi/infra-exec/common"
+	"github.com/mikloslorinczi/infra-exec/executor"
+	"github.com/pkg/errors"
+)
+
+// CompareTags checks if all the tags of the Task matches with the Node's tags.
+func compareTags(taskTags, nodeTags string) bool {
+	taskSplit := strings.Split(taskTags, " ")
+	nodeSplit := strings.Split(nodeTags, " ")
+	for _, taskTag := range taskSplit {
+		found := false
+		for _, nodeTag := range nodeSplit {
+			if taskTag == nodeTag {
+				found = true
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+func getFirstMatch(tasks []common.Task) (common.Task, bool) {
+	for _, task := range tasks {
+		if compareTags(task.Tags, nodeTags) {
+			return task, true
+		}
+	}
+	return common.Task{}, false
+}
+
+func getUnassigned(tasks []common.Task) ([]common.Task, bool) {
+	found := false
+	unassigned := []common.Task{}
+	for _, task := range tasks {
+		if task.Node == "None" {
+			found = true
```

Comment:
Why don't you just return the values from here?
I neither see a reason to continue the loop nor the carry the values when a match has been found.

Reply:
This function aims to select all Tasks which is unassigned (task.Node == "None"), so they can be checked if the Node's Tags are matching with any of them.